libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -884,10 +884,6 @@ float AP_GPS::get_lag(uint8_t instance) const
     if (_delay_ms[instance] > 0) {
         // if the user has specified a non zero time delay, always return that value
         return 0.001f * (float)_delay_ms[instance];
-    } else if (drivers[instance] == nullptr || state[instance].status == NO_GPS) {
-        // the user has not specified a value and we cannot determine it from the GPS type
-        // so return a default delay of 1 measurement interval
-        return 0.001f * (float)_rate_ms[instance];
```

Comment:
If the user has specified a delay for that instance, then we should use it, otherwise, given we don't know what type of GPs it is, we should set the delay to one measurement interval (200msec for a 5Hz GPS).